src/main/java/seedu/subjects/SubjectList.java

```diff
@@ -139,39 +141,76 @@ public static void listSubjects(ArrayList<Subject> subjects) {
      * Adds an event to the deck.
      * @param event Event to be added.
      */
-    public void addEvent(Event event) {
+    public void addEvent(Event event) throws EscException {
         events.add(event);
+        System.out.println(event.getTopic() + " has been added.");
+        listUpcoming();
     }
 
     /**
      * Removes an event from the deck.
      * @param index Index of event to be removed.
+     * @throws EscException if list of events is empty or event item at index does not exit.
      */
     public void removeEvent(int index) throws EscException {
         if (this.events.size() == 0) {
             throw new EscException("The event list is empty.");
         }
         try {
+            System.out.println(events.get(index).getTopic() + " has been removed.");
             events.remove(index);
+            listUpcoming();
         } catch (IndexOutOfBoundsException e) {
             throw new EscException("The event item does not exist.");
         }
     }
 
     /**
      * Lists all upcoming events.
-     * @param events List of upcoming events.
+     * @param dateRange Date range of upcoming events to show.
      */
-    public static void listUpcoming(ArrayList<Event> events) {
-        if (events.size() == 0) {
-            System.out.println("No upcoming events.");
+    public void listUpcoming(int... dateRange) {
```

Comment:
Would it be better to put this & findUpcoming() into the Events package/class instead? But if this is preferable then it's ok

Reply:
Actually thats true. I think I should create an EventList class like the CardList class. So I can shift some methods over also